rest/src/test/java/io/atomix/rest/impl/VertxRestServiceTest.java

```diff
@@ -174,6 +174,156 @@ public void testMessages() throws Exception {
   }
 
+  @Test
+  public void testLock() throws Exception {
+    JsonNodeFactory jsonFactory = JsonNodeFactory.withExactBigDecimals(true);
+    JsonNode json = jsonFactory.objectNode()
+        .put("type", "lock")
+        .set("protocol", jsonFactory.objectNode()
+            .put("type", "multi-primary")
+            .put("backups", 2));
+
+    given()
+        .spec(specs.get(0))
+        .contentType(ContentType.JSON)
+        .body(json)
+        .when()
+        .post("primitives/test-lock")
+        .then()
+        .statusCode(200);
+
+    given()
+        .spec(specs.get(0))
+        .when()
+        .post("primitives/test-lock/lock")
+        .then()
+        .statusCode(200)
+        .body(equalTo("1"));
+
+    given()
+        .spec(specs.get(0))
+        .when()
+        .delete("primitives/test-lock/lock")
+        .then()
+        .statusCode(200);
+  }
+
+  @Test
+  public void testSemaphore() throws Exception {
+    JsonNodeFactory jsonFactory = JsonNodeFactory.withExactBigDecimals(true);
+    JsonNode json = jsonFactory.objectNode()
+        .put("type", "semaphore")
+        .put("initial-capacity", 2)
+        .set("protocol", jsonFactory.objectNode()
+            .put("type", "multi-primary")
+            .put("backups", 2));
+
+    given()
+        .spec(specs.get(0))
+        .contentType(ContentType.JSON)
+        .body(json)
+        .when()
+        .post("primitives/test-semaphore")
+        .then()
+        .statusCode(200);
+
+    given()
+        .spec(specs.get(0))
+        .when()
+        .get("primitives/test-semaphore/permits")
+        .then()
+        .statusCode(200)
+        .body(equalTo("2"));
+
+    given()
+        .spec(specs.get(0))
+        .contentType(ContentType.JSON)
+        .when()
+        .post("primitives/test-semaphore/acquire")
+        .then()
+        .statusCode(200);
+
+    given()
+        .spec(specs.get(1))
+        .contentType(ContentType.JSON)
+        .when()
+        .post("primitives/test-semaphore/acquire")
+        .then()
+        .statusCode(200);
+
+    given()
+        .spec(specs.get(2))
+        .when()
+        .get("primitives/test-semaphore/permits")
+        .then()
+        .body(equalTo("0"));
+
+    given()
+        .spec(specs.get(1))
+        .contentType(ContentType.JSON)
+        .when()
+        .post("primitives/test-semaphore/release")
+        .then()
+        .statusCode(200);
+
+    given()
+        .spec(specs.get(0))
+        .when()
+        .get("primitives/test-semaphore/permits")
+        .then()
+        .body(equalTo("1"));
+
+    given()
+        .spec(specs.get(1))
+        .when()
+        .get("primitives/test-semaphore/permits")
+        .then()
+        .body(equalTo("1"));
+  }
+
+  @Test
+  public void testValue() throws Exception {
+    JsonNodeFactory jsonFactory = JsonNodeFactory.withExactBigDecimals(true);
+    JsonNode json = jsonFactory.objectNode()
+        .put("type", "value")
+        .set("protocol", jsonFactory.objectNode()
```

Comment:
should be protocolConfig right?

Reply:
Should be `protocol`. Shouldn't have removed the property naming strategy